Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -37,34 +36,46 @@ bool LogFileReader::readAll(const QString& fileName)
         if (!parseLine(line, batteryData))
         {
             qDebug() << "Error while parsing" << line;
-            // return false;
         }
         else
         {
-            // This is how to send out a signal in QT using the emit keyword.
-            // This line notifies the classes listening to this signal
-            // that battery data has been received.
             emit batteryDataReceived(batteryData);
         }
     }
 
     return true;
 }
 
-/* File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
- * Need to implement error checking for the correct number of values and
- * that the conversion from string to double is sucessful.*/
+
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     QStringList sections = line.split(BATDATA_DELIMITER);
 
+    if (sections.length() != COLUMNS)
+    {
+        return false;
+    }
+
     QString timeString = sections.at(0);
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    bool voltageOk;
+    bool currentOk;
 
-    batteryData.current = sections.at(2).toDouble();
+    batteryData.voltage = sections.at(1).toDouble(&voltageOk);
 
-    return true;
+    batteryData.current = sections.at(2).toDouble(&currentOk);
 
+    if (!voltageOk || !currentOk)
+    {
+        return false;
+    }
```

Comment:
Let's add a new line here (Between 72 and 73). Since typically one wouldn't put a new line between `else` and `else if` statements, this `if` statement can look a bit like an `else if` one.